orgparse/__init__.py

```diff
@@ -130,16 +129,31 @@ def load(path: Union[str, Path, TextIO], env: Optional[OrgEnv]=None) -> OrgNode:
 
     """
     orgfile: TextIO
-    if isinstance(path, (str, Path)):
-        # Use 'with' to close the file inside this function.
-        with codecs.open(str(path), encoding='utf8') as orgfile:
-            lines = (l.rstrip('\n') for l in orgfile.readlines())
-        filename = str(path)
-    else:
-        orgfile = path
-        lines = (l.rstrip('\n') for l in orgfile.readlines())
-        filename = path.name if hasattr(path, 'name') else '<file-like>'
-    return loadi(lines, filename=filename, env=env)
+
+    orgfile = path
+
+    # file-like object (e.g. io.StringIO)
+    try:
```

Comment:
All `if` are gone and replaced by more resource-efficient `try...except`.
It could be discussed if this improve the readability of the code or not. IMHO it doesn't affect readability negativ in that special case here.

Reply:
hm, I appreciate you are meaning well and want to improve, but we won't get far with this approach :) 

First, I'm not sure I agree try/except are more resource efficient even from theoretical perspective (you need to maintain unwinding stack etc). But either way, in this case: 

1) it's definitely not clear which one is faster -- it's only possible to find out unless you microbenchmark this specific bit of code
2) even if try/catch is faster, it's going to be such a tiny amount, comparing to IO operations like `.readlines`, that it's just a waste of time to optimize
3) try/catch definitely ruins readability, it's harder to reason about exceptions and their types
4) try/catch makes code less safe -- e.g. what if something else throws AttributeError? E.g. if you make a typo in `.rstrip`. If you're willing to go with try/except path you need to wrap as little code in try/except as possible; and then it would hurt readability even more since you'll have to use `else:` block
5) it messes with mypy (since it can't infer the type information anymore), again making code less safe

I appreciate thinking about performance, but there are plenty of other opportunities in this library where it would make much more sense without hurting the readability.